### DISCUSSION TITLE
Is there a way to have code splitting with createReducer, createActions, createSlice?

### DISCUSSION THREAD

**Discussion by marcusjwhelan** [reduxjs/redux-toolkit#2587] [General]
Our application is pretty large and we use a lot of redux toolkit tools but would like to make sure that the first load doesn't load all reducers, actions, and slices which is actually a lot of code. It would be preferable that certain routes load certain slices. 

And possibly Reducers/Actions once they are needed.

**Comment by markerikson** [comment]
Yes, we cover this in the core docs:

https://redux.js.org/usage/code-splitting

Also, while it's not anything official, there's a discussion about using Suspense to help with the process of loading a reducer for a given component subtree and making sure it doesn't render until that data is available:

https://github.com/reduxjs/redux/discussions/4114
